src/components/Ideas/CompIdeaDetail.js

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useEffect, useState } from "react"
+import { IdeaContext } from "./IdeaProvider"
+import { useParams, useHistory } from "react-router-dom"
+import "./Idea.css"
+
+export const CompIdeaDetail = (props) => {
+    const { ideas, getCompleteIdeas, deleteIdea } = useContext (IdeaContext)
+    const [ idea, setIdea ] = useState({ artist: {} })
+
+    const { ideaId } = useParams()
+
+    useEffect(() => {
+        if(props.idea){
+            setIdea(props.idea)
+        } else {
+            const thisIdea = idea.find(i => i.id === parseInt(ideaId)) || { artist: {} }
```

Comment:
If this is coming from the database, it's better to make the API call instead of filtering here. It makes for better performance. Also what if there were 1,000 ideas... There's a possibility that it could have to hit each and everyone until it finds the correct item.

Reply:
Also, if this is an array, `idea` should change to `ideas` or `ideaArray`, or something like that.